### ISSUE TITLE
VS Code crashes with reconversion from google IME input

### ISSUE THREAD

**Issue by tamuratak** [microsoft/vscode#230303]
Version: 1.94.0-insider
Commit: 889feaf2ef663d9d3b079c4a3627e4ce27027323
Date: 2024-10-02T04:46:15.711Z
Electron: 30.5.1
ElectronBuildId: 10262041
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Darwin arm64 22.6.0

Steps to Reproduce:

1. Set `"editor.experimentalEditContextEnabled": true`
2. Enable Google IME for Japanese.
3. Type `sennsei`
4. Type `Space` twice
5. Click a different line of the editor with the mouse.
6. Press the key for reconversion,  the `Kana` key or `Ctrl-Shift-R`.
7. VS Code Insiders crashes sometimes, not always. It would be difficult to reproduce.

The following is the crash dump:

[fce4fc0f-f989-4de4-a9d7-4b2ace9c8859.dmp](https://github.com/user-attachments/files/17229566/fce4fc0f-f989-4de4-a9d7-4b2ace9c8859.dmp)

**Comment by aiday-mar** [comment]
Hi @tamuratak may I ask you if you still see this issue happening 6 months later? We have done a lot of changes to the edit context since then which maybe has fixed this

**Comment by tamuratak** [comment]
I can still reproduce the issue. I will upload a new crash dump if I can.

**Comment by tamuratak** [comment]
No, the reconversion doesn't work when `editor.experimentalEditContextEnabled: false`.

**Comment by aiday-mar** [comment]
Hi thanks for the clarification! It would seem this is linked to an upstream chromium issue!

**Comment by deepak1556** [comment]
@tamuratak we would like to confirm fixes from the runtime but we have not been able to repro this issue so far. Can you confirm if I am missing or performing incorrect steps in the attached recording.

https://github.com/user-attachments/assets/56297d51-7f41-4ea4-a64d-8bf25ed5bb67

**Comment by tamuratak** [comment]
You are using macOS IME for Japanese. I cannot reproduce the issue with it.

With Google IME for Japanese, I can reproduce the issue.

**Comment by deepak1556** [comment]
Ah that was it, thanks for clarifying.
